### PR TITLE
IOS-12102: fix failing to open sharedlink items' subdirs/file levels if password required

### PR DESCRIPTION
**Pull request by thuyBox**

### CODE REVIEW

**Comment by boxcla** [comment]
Hi @thuyBox, thanks for the pull request. Before we can merge it, we need you to sign our Contributor License Agreement. You can do so electronically here: http://opensource.box.com/cla

Once you have signed, just add a comment to this pull request saying, "CLA signed". Thanks!

**Comment by thuyBox** [comment]
CLA signed

**Comment by boxcla** [comment]
Verified that @thuyBox has just signed the CLA. Thanks, and we look forward to your contribution.

File: BoxContentSDK/BoxContentSDK/Helper/BOXSharedLinkHeadersHelper.m [hunk 1]
```diff
@@ -60,9 +60,10 @@ - (void)storeHeadersFromAncestorsIfNecessaryForItemWithID:(NSString *)itemID ite
         BOXFolderMini *folder = ancestors[i];
             
         NSString *link = [self.delegate sharedLinkForItemWithID:folder.modelID itemType:folder.type userID:self.userID];
+        NSString *password = [self.delegate passwordForSharedItemWithID:folder.modelID itemType:folder.type userID:self.userID];
```
Comment (boxtcarpel) [inline BoxContentSDK/BoxContentSDK/Helper/BOXSharedLinkHeadersHelper.m]: Can move this inside the if statement.

**Comment by boxtcarpel** [comment]
+2

LGTM!

🚈

**Pull request merged**